src/Elastic.Apm.MongoDb/Elastic.Apm.MongoDb.csproj

```diff
@@ -0,0 +1,22 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>net461;netstandard2.0</TargetFrameworks>
+    <RootNamespace>Elastic.Apm.MongoDb</RootNamespace>
+    <AssemblyName>Elastic.Apm.MongoDb</AssemblyName>
+    <PackageId>Elastic.Apm.MongoDb</PackageId>
+    <Description>
+      Elastic APM integration with the MongoDB.Driver package.
+    </Description>
+    <IsPackable>true</IsPackable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="MongoDB.Driver.Core" Version="2.4.4" />
```

Comment:
Should this be set to a Version range, up to 3.0.0 initially? e.g. 

```
<PackageReference Include="MongoDB.Driver.Core" Version="[2.4.4,3.0)" />
```

Reply:
Why would we do that? That way we'd not work with newer versions even the code would work otherwise. This PR only depends on `MongoDB.Driver.Core.Events.IEventSubscriber`. I'd be optimistic and think that it's more likely that this will not change in later versions. So I'd take a more optimistic approach here, otherwise we'll always need to update these when new versions get released.

Reply:
It is not known at this point in time whether this will work for _all_ future versions of MongoDB.Driver.Core, so this is just a cautious suggestion. It would only need to be updated when a new major version is released, which is not very frequent.

Reply:
Currently we don't do this with any of the supported libraries - so far the approach is to depend on the lowest possible version and not limit support. The main thinking behind this is that we usually only depend on a few basic APIs - in this case it's a single interface.

I feel we should be consistent here - either we purposely limit support for current versions and proactively watch new releases and check compatibility for all our supported libraries, or we keep the current approach. But mixing them does not feel good. If we limit it now, we should have some strategy to make sure once `3.0` is out we know about it and we release immediately. It'd be bad to wait for users to report.

If we move to this approach then we'll need to
- actively check for new versions
- when a new version comes out we should test
- update the version of the dependency and release a new version

In my opinion we'd take on a huge maintenance work and so far we haven't seen incompatibility issues in newer versions, so I feel the benefit would be limited.

I agree with your statement
> It is not known at this point in time whether this will work for all future versions 

But if we want to tackle this issue (which holds true for all our supported libraries) then first we need to have a system to proactively release for each new version of the supported library, otherwise we'd always be behind and users will ping us to update in order to support newer versions. Also, based on the experience I think in most of the cases we'll only need to update the version range, because changes that break our code are fairly rare.

Reply:
We discussed this in the last sync-meeting: for now we keep the support open-ended and deal with this once this really causes a problem.